### DISCUSSION TITLE
Comparing Models With and Without Spatio-Temporal Random Effects

### DISCUSSION THREAD

**Discussion by Nathan-Hebert** [sdmTMB/sdmTMB#236] [Q&A] [answered]
Is there a way to quantitatively compare the amount of variability explained by a model incorporating spatio-temporal random effects + fixed effects versus a model containing only fixed effects? I know I can plot predictions and conduct cross-validation, but is there any sort of in-sample metric I can derive to supplement these approaches?

As an example, I fit spatial and non-spatial models in mgcv recently, and I could compare the deviance explained by the two versions.

Thanks!

**Comment by seananderson** [comment, marked as answer]
@Nathan-Hebert, sorry I completely missed this question back in July! This work on an R2 function might be what you're looking for: https://github.com/pbs-assess/sdmTMB/blob/main/scratch/r2-glmm.qmd within [this](https://github.com/pbs-assess/sdmTMB/issues/256) issue.

Not all families are in there yet and it's not thoroughly tested, but that might be a start.

**Comment by Nathan-Hebert** [comment]
Awesome, thank you!
